### PR TITLE
WT-8448 - s_mentions finds content that should not be matching

### PR DESCRIPTION
**Pull request by ajmorton**
Generated build scripts in `build_posix` and `cmake_build` will contain wiredtiger's filepath, and when the filepath contains the ticket that `s_mentions` is checking for (e.g. `~/src/wt-8448/wiredtiger`) then `s_mentions` will return false positive matches.

Update `s_mentions` to handle this case.

Additionally ensure that any reported matches by `s_mentions` are treated as warnings only and `s_all` does not fail when they are detected.

### CODE REVIEW

**Comment by ajmorton** [comment]
This has been tested to work against the following folder paths:
```
src/wt-8448/wiredtiger
src/WT-8448/wiredtiger
src/wt-8448-s_mentions-false-positive/wiredtiger
src/wt-8448_wt_renamed
```

@sulabhM can you try these changes on your local clone? I wasn't able to replicate the exact contents of `config.log` mentioned in the ticket.

File: dist/s_mentions [hunk 1]
```diff
@@ -7,17 +7,32 @@
...
+if [[ $wt_src_path =~ .*${wt_ticket_regex}.* ]]; then
+    # If the wiredtiger filepath contains $ticket_id (e.g. ~/src/$ticket_id/wiredtiger) then we will
+    # match generated build scripts containing this path. Filter these results from the search.
+    search_function="$search_function | grep -v $wt_src_path"
+fi
+
 # Check for comments related to the ticket.
-if grep -inr --exclude-dir=.git "$ticket_id" ../. > /dev/null 2>&1; then
-    echo "There are comments mentioning $ticket_id in the code, please check them."
+if eval "$search_function >/dev/null" ; then
```
Comment (etienneptl) [inline dist/s_mentions:32]: We still need `2>&1`

File: dist/s_mentions [hunk 2]
```diff
@@ -7,17 +7,32 @@
...
+    echo "There are comments mentioning $ticket_id in the code, please check if they need to be \
+resolved."
+    eval "$search_function -m 10"
```
Comment (etienneptl) [inline dist/s_mentions:35]: I don't think limiting the number of results is useful, we want to see where the references are.

**Comment by sulabhM** [comment]
I tried from the original path I was hitting this issue. I get this now:

```
sulabh@sud:~/work/wt-8306-resolve-upds/dist$ ./s_all
dist/s_all run started...
Updating files that include the package version
####################### MESSAGE ############################
s_all run of: "./s_mentions" resulted in:
    There are comments mentioning wt-8448 in the code, please check if they need to be resolved.
    ../dist/s_mentions:22:    # If the wiredtiger filepath contains $ticket_id (e.g. ~/src/wt-8448/wiredtiger) then we will match a lot of
    ../dist/__wt.4079:1:There are comments mentioning wt-8448 in the code, please check if they need to be resolved.
#######################
dist/s_all run finished
```

**Comment by ajmorton** [comment]
Thanks for checking! 
`s_mentions` extracts the WT ticket to search for from the git branch, and since you've checked out `wt-8448-s_mentions_false_positive` it's looking for WT-8448 in the code instead of WT-8306. creating a temp wt-8306* branch will search for 8306 again.

I had a look at the contents of dist/filelist but it only includes C files. I think we want to keep the search list as broad as possible so we can catch FIXMEs in files like `evergreen.yml` as well.

**Review by etienneptl — APPROVED** [review]
LGTM

**Comment by sulabhM** [comment]
> I had a look at the contents of dist/filelist but it only includes C files. I think we want to keep the search list as broad as possible so we can catch FIXMEs in files like evergreen.yml as well.

Cool, lets go with and we can rework if someone gets blocked with the new behavior.

**Pull request merged**